pkg/controller/workspace/provision/deployment.go

```diff
@@ -53,11 +54,12 @@ func SyncDeploymentToCluster(
 	workspace *v1alpha1.Workspace,
 	podAdditions []v1alpha1.PodAdditions,
 	saName string,
-	clusterAPI ClusterAPI) DeploymentProvisioningStatus {
+	clusterAPI ClusterAPI,
+	isPVCRequired bool) DeploymentProvisioningStatus {
```

Comment:
Considering that `isPVCRequired` is in the `provision` package, and depends only on the workspace spec, we can just repeat the check where it's needed (i.e. in `getSpecDeployment`). This would keep the signature for `SyncDeploymentToCluster` simpler and would allow `workspace_controller.go` to be simplified
```
	if provision.IsPVCRequired(workspace.Spec.Devfile.Components) {
		provision.GeneratePVC(workspace, r.client, reqLogger)
	}
```